model/scratch/dffml_model_scratch/model/slr.py

```diff
@@ -0,0 +1,160 @@
+        super().__init__(parent, features)
+        self.xData = np.array([])
+        self.yData = np.array([])
+        self.features = self.applicable_features(features)
+        self._features_hash_ = hashlib.sha384((''.join(sorted(self.features))).encode()).hexdigest()
+
+    @property
+    def regression_line(self):
+        return self.parent.saved.get(self._features_hash_, None)
+
+    @regression_line.setter
+    def regression_line(self, rline):
+        self.parent.saved[self._features_hash_] = rline
+
+    def applicable_features(self, features):
+        usable = []
+        if len(features) != 1:
+            raise ValueError("Simple Linear Regression doesn't support features other than 1")
+        for feature in features:
+            if feature.dtype() != int and feature.dtype() != float:
+                raise ValueError("Simple Linear Regression only supports int or float feature")
+            if feature.length() != 1:
+                raise ValueError("Simple LR only supports single values (non-matrix / array)")
+            usable.append(feature.NAME)
+        return sorted(usable)
+
+    async def predict_input(self, x):
+        prediction = self.regression_line[0]*x+self.regression_line[1]
+        self.logger.debug("Predicted Value of {} {}:".format(self.parent.config.predict, prediction))
+        return prediction
+
+    async def squared_error(self, ys, yline):
+        return sum((ys - yline)**2)
+
+    async def coeff_of_deter(self, ys, regression_line):
+        y_mean_line = [np.mean(ys) for y in ys]
+        squared_error_mean = await self.squared_error(ys, y_mean_line)
+        squared_error_regression = await self.squared_error(ys, regression_line)
+        return 1 - (squared_error_regression/squared_error_mean)
+
+    async def best_fit_line(self):
+        self.logger.debug("Number of input repos: {}".format(len(self.xData)))
+        x = self.xData
+        y = self.yData
+        mean_x = np.mean(self.xData)
+        mean_y = np.mean(self.yData)
+        m = (mean_x*mean_y - np.mean(x*y))/((mean_x**2) - np.mean(x*x))
+        b = mean_y - (m*mean_x)
+        regression_line = [m*x + b for x in x]
+        accuracy = await self.coeff_of_deter(y, regression_line)
+        return (m, b, accuracy)
+
+    async def train(self, sources: Sources):
+        async for repo in sources.with_features(self.features +
+                                                [self.parent.config.predict]):
+            feature_data = repo.features(self.features +
+                                         [self.parent.config.predict])
+            self.xData = np.append(self.xData, feature_data[self.features[0]])
+            self.yData = np.append(self.yData, feature_data[self.parent.config.predict])
+        self.regression_line = await self.best_fit_line()
+
+    async def accuracy(self, sources: Sources) -> Accuracy:
+        if self.regression_line is None:
+            raise ValueError('Model Not Trained')
```

Comment:
Will merge this in current state once CI tests pass will change ValueError after #125 is fixed